lib/resque-heroku-signals.rb

```diff
@@ -12,24 +12,43 @@ def self.heroku_will_terminate?
 #   https://github.com/resque/resque/blame/v2.0.0/lib/resque/worker.rb#L406
 # https://github.com/resque/resque/issues/1559#issuecomment-310908574
 Resque::Worker.class_eval do
-  def unregister_signal_handlers
-    trap('TERM') do
-      $HEROKU_WILL_TERMINATE_RESQUE = true
+  # In this patched implementation, the only change is that the worker sends
+  # SIGINT to the child, the rest is copied verbatim.
+  def new_kill_child
+    if @child
+      unless child_already_exited?
+        if pre_shutdown_timeout && pre_shutdown_timeout > 0.0
+          log_with_severity :debug, "Waiting #{pre_shutdown_timeout.to_f}s for child process to exit"
+          return if wait_for_child_exit(pre_shutdown_timeout)
+        end
 
-      trap('TERM') do
-        log_with_severity :info, "[resque-heroku] received second term signal, throwing term exception"
+        log_with_severity :debug, "Sending INT signal to child #{@child}"
+        Process.kill("INT", @child)
```

Comment:
👍 . for a reference, https://github.com/resque/resque/blob/e5e03962c297bbc9392e17bfccb67799d17c7778/lib/resque/worker.rb#L543-L566 .

INT over TERM.